src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name, health) {
+    super(name, health);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name, health) {
+    super(name, health);
```

Comment:
[FUNCTION NAMING] - The constructor of the `Carnivore` class should not accept a `health` parameter for the same reason as the `Herbivore` class. The health should be `100` by default. Please remove the `health` parameter from the constructor.